File: python-learning/challenges/reverse_string.py
## reverse string challenge
# Write a function that takes a string as input and returns the string reversed.


def reverse_string(text: str) -> str:
    return text[::-1]


print(reverse_string("Hello World"))


def reverse_string_hard(text):
    reversed = ""

    for i in range(len(text) - 1, -1, -1):
        reversed += text[i]

    return "".join(reversed)


print(reverse_string_hard("Hello fucking World"))

File: dsa/3-arrays/challenges/01-reverse-string/reverseString.js
function reverseString(text) {
    if (typeof text != "string") throw new Error('Input must be a string')
    if (text.length === 0) throw new Error('String cannot be empty')
    const reversed = text.split('').reverse().join('')
    console.log(reversed)
}

reverseString('Hello')